### PR TITLE
New Resource: aws_imagebuilder_component

### LINKED ISSUE #13315: Feature Request: EC2 ImageBuilder Component
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In December 2019 AWS announced the availability of EC2 ImageBuilder, a service that makes it easier and faster to build and maintain secure images. Image Builder simplifies the creation, patching, testing, distribution, and sharing of Linux or Windows Server images. 

### New Resource

* aws_imagebuilder_component

### Potential Terraform Configuration

```hcl
resource "aws_imagebuilder_component" "example" {
  arn                = ""
  client_token       = ""
  change_description = ""
  data               = ""
  description        = ""
  encrypted          = true 
  kms_key_id         = ""
  name               = ""
  owner              = "" 
  platform           = ""
  tags               = {}
  type               = ""
  uri                = "" 
  version            = "" 
}
```

### References

Announcement: https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-ec2-image-builder/
API Reference: https://docs.aws.amazon.com/imagebuilder/latest/APIReference/Welcome.html
SDK Reference: https://docs.aws.amazon.com/sdk-for-go/api/service/imagebuilder/
* #11084

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: #11084
Closes #13315
Reference: https://github.com/hashicorp/terraform-provider-aws/pull/13485

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
* **New Data Source:** `aws_imagebuilder_component`
* **New Resource:** `aws_imagebuilder_component`
```

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAwsImageBuilderComponent_basic (18.98s)
--- PASS: TestAccAwsImageBuilderComponent_ChangeDescription (19.15s)
--- PASS: TestAccAwsImageBuilderComponent_Description (19.29s)
--- PASS: TestAccAwsImageBuilderComponent_disappears (15.10s)
--- PASS: TestAccAwsImageBuilderComponent_KmsKeyId (23.58s)
--- PASS: TestAccAwsImageBuilderComponent_Platform_Windows (19.22s)
--- PASS: TestAccAwsImageBuilderComponent_SupportedOsVersions (19.15s)
--- PASS: TestAccAwsImageBuilderComponent_Tags (42.96s)
--- PASS: TestAccAwsImageBuilderComponent_Uri (35.66s)

--- PASS: TestAccAwsImageBuilderComponentDataSource_Arn (18.51s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAwsImageBuilderComponent_basic (24.13s)
--- PASS: TestAccAwsImageBuilderComponent_ChangeDescription (23.39s)
--- PASS: TestAccAwsImageBuilderComponent_Description (24.43s)
--- PASS: TestAccAwsImageBuilderComponent_disappears (19.28s)
--- PASS: TestAccAwsImageBuilderComponent_KmsKeyId (28.49s)
--- PASS: TestAccAwsImageBuilderComponent_Platform_Windows (24.54s)
--- PASS: TestAccAwsImageBuilderComponent_SupportedOsVersions (24.05s)
--- PASS: TestAccAwsImageBuilderComponent_Tags (53.99s)
--- PASS: TestAccAwsImageBuilderComponent_Uri (40.78s)

--- PASS: TestAccAwsImageBuilderComponentDataSource_Arn (24.26s)
```

Output from sweeper in AWS Commercial:

```
2020/11/12 16:58:32 [DEBUG] Running Sweepers for region (us-west-2):
2020/11/12 16:58:32 [DEBUG] Running Sweeper (aws_imagebuilder_component) in region (us-west-2)
2020/11/12 16:58:45 Sweeper Tests ran successfully:
  - aws_imagebuilder_component
2020/11/12 16:58:45 [DEBUG] Running Sweepers for region (us-east-1):
2020/11/12 16:58:45 [DEBUG] Running Sweeper (aws_imagebuilder_component) in region (us-east-1)
2020/11/12 16:58:46 Sweeper Tests ran successfully:
  - aws_imagebuilder_component
ok    github.com/terraform-providers/terraform-provider-aws/aws 18.223s
```
Output from sweeper in AWS GovCloud (US):

```
2020/11/12 16:59:00 [DEBUG] Running Sweepers for region (us-gov-west-1):
2020/11/12 16:59:00 [DEBUG] Running Sweeper (aws_imagebuilder_component) in region (us-gov-west-1)
2020/11/12 16:59:03 Sweeper Tests ran successfully:
  - aws_imagebuilder_component
ok    github.com/terraform-providers/terraform-provider-aws/aws 5.004s
```

### CODE REVIEW

**Review by gdavison — APPROVED** [review]
LGTM 🚀 

A couple cases of `fmt.Errorf("...%s", ..., err)`

Commercial partition:

--- PASS: TestAccAwsImageBuilderComponent_disappears (42.98s)
--- PASS: TestAccAwsImageBuilderComponent_KmsKeyId (56.76s)
--- PASS: TestAccAwsImageBuilderComponent_SupportedOsVersions (57.21s)
--- PASS: TestAccAwsImageBuilderComponentDataSource_Arn (57.05s)
--- PASS: TestAccAwsImageBuilderComponent_basic (57.16s)
--- PASS: TestAccAwsImageBuilderComponent_Description (58.27s)
--- PASS: TestAccAwsImageBuilderComponent_ChangeDescription (58.43s)
--- PASS: TestAccAwsImageBuilderComponent_Platform_Windows (58.96s)
--- PASS: TestAccAwsImageBuilderComponent_Uri (59.75s)
--- PASS: TestAccAwsImageBuilderComponent_Tags (85.60s)

GovCloud partition:

--- PASS: TestAccAwsImageBuilderComponent_disappears (19.97s)
--- PASS: TestAccAwsImageBuilderComponent_Description (26.93s)
--- PASS: TestAccAwsImageBuilderComponentDataSource_Arn (27.72s)
--- PASS: TestAccAwsImageBuilderComponent_Platform_Windows (28.48s)
--- PASS: TestAccAwsImageBuilderComponent_SupportedOsVersions (29.18s)
--- PASS: TestAccAwsImageBuilderComponent_ChangeDescription (29.06s)
--- PASS: TestAccAwsImageBuilderComponent_KmsKeyId (29.20s)
--- PASS: TestAccAwsImageBuilderComponent_basic (29.30s)
--- PASS: TestAccAwsImageBuilderComponent_Uri (29.53s)
--- PASS: TestAccAwsImageBuilderComponent_Tags (42.20s)

File: aws/data_source_aws_imagebuilder_component.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	d.Set("description", component.Description)
+	d.Set("encrypted", component.Encrypted)
+	d.Set("kms_key_id", component.KmsKeyId)
+	d.Set("name", component.Name)
+	d.Set("owner", component.Owner)
+	d.Set("platform", component.Platform)
+	d.Set("supported_os_versions", aws.StringValueSlice(component.SupportedOsVersions))
+
+	if err := d.Set("tags", keyvaluetags.ImagebuilderKeyValueTags(component.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
```
Comment (bflad) [inline aws/data_source_aws_imagebuilder_component.go:113]:
```suggestion
		return fmt.Errorf("error setting tags: %w", err)
```

File: aws/resource_aws_imagebuilder_component.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+	d.Set("description", component.Description)
+	d.Set("encrypted", component.Encrypted)
+	d.Set("kms_key_id", component.KmsKeyId)
+	d.Set("name", component.Name)
+	d.Set("owner", component.Owner)
+	d.Set("platform", component.Platform)
+	d.Set("supported_os_versions", aws.StringValueSlice(component.SupportedOsVersions))
+
+	if err := d.Set("tags", keyvaluetags.ImagebuilderKeyValueTags(component.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
```
Comment (bflad) [inline aws/resource_aws_imagebuilder_component.go:214]:
```suggestion
		return fmt.Errorf("error setting tags: %w", err)
```

**Review by bflad — COMMENTED** [review]
Fixing the ones I see

**Pull request merged**